### ISSUE TITLE
Errors using MCP > Tag Creator

### ISSUE THREAD

**Issue by ksurendra** [Adobe-Consulting-Services/acs-aem-commons#1301]
## Required Information

* [ ] AEM Version, including Service Packs, Cumulative Fix Packs, etc: AEM 6.3
* [ ] ACS AEM Commons Version: 3.14.10
* [ ] Reproducible on Latest? Yes

### Expected Behavior

Allow to upload an xlsx file to "Tag Creator" process to generate Tags.

### Actual Behavior

Process not linking the attached xlsx file. And always seeing a error or task failed.

### Steps to Reproduce

1. Manage Controlled Process > Start Process > Tag Creator
2. Choose a pre-formatted xlsx file (attached to this document) that has tags' information and attach it (even you do not choose the file and just hit "Start", the same error happens)
3. Choose any option under "Primary Converter" or "Fallback Converter" 
4. Choose "Start"

The report section shows that there is an "error".

**Sample input file used**
[nfl1.xlsx](https://github.com/Adobe-Consulting-Services/acs-aem-commons/files/1869230/nfl1.xlsx)

-- NOTE: The same xlsx worked while using ACS AEM Tools > Tag Maker

** Error screenshots**
![empty-report](https://user-images.githubusercontent.com/902972/38214548-997eec0a-368a-11e8-87fd-1f9544bbb22d.png)
![reports](https://user-images.githubusercontent.com/902972/38214550-99a1506a-368a-11e8-9a22-2219f8be2025.png)

**Snippet from Error log:**

```
02.04.2018 15:19:38.830 *ERROR* [pool-308-thread-1] com.adobe.acs.commons.fam.impl.ActionManagerImpl Error in error handler for action Tag Creator: Create tags using [ Title And Node Name -> Lowercase With Dashes ]: Parse tags (0e27bdf6-d391-4c43-b39b-737bf5a46628)
java.lang.NullPointerException: null
	at org.apache.poi.openxml4j.opc.internal.ZipHelper.prepareToCheckHeader(ZipHelper.java:211)
	at org.apache.poi.openxml4j.opc.internal.ZipHelper.openZipStream(ZipHelper.java:228)
	at org.apache.poi.openxml4j.opc.ZipPackage.<init>(ZipPackage.java:97)
	at org.apache.poi.openxml4j.opc.OPCPackage.open(OPCPackage.java:342)
	at org.apache.poi.util.PackageHelper.open(PackageHelper.java:37)
	at org.apache.poi.xssf.usermodel.XSSFWorkbook.<init>(XSSFWorkbook.java:285)
	at com.adobe.acs.commons.mcp.impl.processes.TagCreator.lambda$parseTags$0(TagCreator.java:127)
	at com.adobe.acs.commons.fam.impl.ActionManagerImpl.withResolver(ActionManagerImpl.java:188)
	at com.adobe.acs.commons.mcp.impl.processes.TagCreator.parseTags(TagCreator.java:126)
	at com.adobe.acs.commons.mcp.impl.ProcessInstanceImpl.lambda$runStep$13(ProcessInstanceImpl.java:219)
	at com.adobe.acs.commons.fam.impl.ActionManagerImpl.withResolver(ActionManagerImpl.java:188)
	at com.adobe.acs.commons.fam.impl.ActionManagerImpl.lambda$deferredWithResolver$0(ActionManagerImpl.java:156)
	at com.adobe.acs.commons.fam.impl.TimedRunnable.run(TimedRunnable.java:80)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
02.04.2018 15:19:38.830 *ERROR* [pool-308-thread-1] com.adobe.acs.commons.fam.impl.ActionManagerImpl Caught exception in task: null
java.lang.NullPointerException: null
	at org.apache.poi.openxml4j.opc.internal.ZipHelper.prepareToCheckHeader(ZipHelper.java:211)
	at org.apache.poi.openxml4j.opc.internal.ZipHelper.openZipStream(ZipHelper.java:228)
	at org.apache.poi.openxml4j.opc.ZipPackage.<init>(ZipPackage.java:97)
	at org.apache.poi.openxml4j.opc.OPCPackage.open(OPCPackage.java:342)
	at org.apache.poi.util.PackageHelper.open(PackageHelper.java:37)
	at org.apache.poi.xssf.usermodel.XSSFWorkbook.<init>(XSSFWorkbook.java:285)
	at com.adobe.acs.commons.mcp.impl.processes.TagCreator.lambda$parseTags$0(TagCreator.java:127)
	at com.adobe.acs.commons.fam.impl.ActionManagerImpl.withResolver(ActionManagerImpl.java:188)
	at com.adobe.acs.commons.mcp.impl.processes.TagCreator.parseTags(TagCreator.java:126)
	at com.adobe.acs.commons.mcp.impl.ProcessInstanceImpl.lambda$runStep$13(ProcessInstanceImpl.java:219)
	at com.adobe.acs.commons.fam.impl.ActionManagerImpl.withResolver(ActionManagerImpl.java:188)
	at com.adobe.acs.commons.fam.impl.ActionManagerImpl.lambda$deferredWithResolver$0(ActionManagerImpl.java:156)
	at com.adobe.acs.commons.fam.impl.TimedRunnable.run(TimedRunnable.java:80)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)`
```

### Links

_Links to related assets, e.g. content packages containing test components_

**Comment by davidjgonzalez** [comment]
@ksurendra This xlsx works fine for me ... Are you on Windows? Also, you are uploading a real XLSX file to this and NOT a CSV right? (ACS Tools Tag Maker requires a CSV; this takes an Excel as its more robust)

![image](https://user-images.githubusercontent.com/1451868/38285811-fc75c3b4-378f-11e8-9bd5-1d65dbf4ab03.png)

[tag_creator_no_description.xlsx](https://github.com/Adobe-Consulting-Services/acs-aem-commons/files/1874166/tag_creator_no_description.xlsx)

**Comment by davidjgonzalez** [comment]
@ksurendra can we close this? Can't tell if you were uploading a  CSV (which AFIAK the file upload widget shouldn't have allowed you to do based on the mime type restriction?)

**Comment by ksurendra** [comment]
@davidjgonzalez  I'm using the xlsx and not csv. Anytime i try to attach a file to any MCP process i'm seeing the below exception. But this may not be related to Tag Creator and related to ACS Commons/MCP. You may close this.

`04.04.2018 11:06:43.488 *ERROR* [pool-308-thread-1] com.adobe.acs.commons.fam.impl.ActionManagerImpl Error in error handler for action Tag Creator: Create tags using [ Title And Node Name -> Lowercase With Dashes ]: Parse tags (b69f510c-57dc-4cdd-8c5b-447a250c9f1b)
java.lang.NullPointerException: null`

**Comment by clooky** [comment]
Did anyone make ony progress on this?  I have excatly the same issue .  

I'm on 

- Windows 10
- Java 1.8
- AEM 6.3 SP2
- ACS Commons 3.15.2
- Excel 2016
- .xlsx file extension

When I rename the .xlsx to .zip and view archive in 7zip shows correct format, so this is a zip type excel file.

I get exactlty the same error messages and response.  I have tried my own xlsx file and also the one provided by Suren. I have also tried other MCP processes and get the same behavior reading the header of the xlsx file.

**Comment by ksurendra** [comment]
@clooky i was using a CSV and was the issue. Can you share the sample xlsx you are using. Let me try.

**Comment by badvision** [comment]
With the better error handling do we get more insight into what's going on when this happens?

**Comment by badvision** [comment]
One thing we might consider doing if this latest issue is related to using CSV is to:
- Close this ticket as Not a Bug
- Open a new feature request to support CSV and Tab-delimited files in the Spreadsheet API
- Refactor Tag Creator to use the new Spreadsheet API as it has ample unit test coverage and a lot of flexibility, if it fits.
- I would say the same for Asset Folder Creator but it has a very unique way of handling input which Spreadsheet doesn't accommodate at this time.

**Comment by Lrochefort** [comment]
I have the same problem as @clooky but my setting's a bit different.

I'm on:

- Windows 10 64bits
- Java 1.8
- AEM 6.2 SP1
- ACS Commons 3.17.0
- Excel Office 365
- .xlsx file extension

Trace:

`07.06.2018 16:51:11.244 *ERROR* [pool-647-thread-1] com.adobe.acs.commons.fam.impl.ActionManagerImpl Error in action Tag Creator: Create tags using [ Title And Node Name -> Lowercase With Dashes ]: Parse tags (ae7e19aa-bc57-4d4a-afb2-036df77b85f8)
java.lang.NullPointerException: in is null
        at java.util.zip.ZipInputStream.<init>(Unknown Source)
        at java.util.zip.ZipInputStream.<init>(Unknown Source)
        at org.apache.poi.openxml4j.opc.ZipPackage.<init>(ZipPackage.java:88)
        at org.apache.poi.openxml4j.opc.OPCPackage.open(OPCPackage.java:272)
        at org.apache.poi.util.PackageHelper.open(PackageHelper.java:37)
        at org.apache.poi.xssf.usermodel.XSSFWorkbook.<init>(XSSFWorkbook.java:258)
        at com.adobe.acs.commons.mcp.impl.processes.TagCreator.lambda$parseTags$0(TagCreator.java:131)
        at com.adobe.acs.commons.fam.impl.ActionManagerImpl.withResolver(ActionManagerImpl.java:204)
        at com.adobe.acs.commons.mcp.impl.processes.TagCreator.parseTags(TagCreator.java:130)
        at com.adobe.acs.commons.mcp.impl.ProcessInstanceImpl.lambda$runStep$14(ProcessInstanceImpl.java:238)
        at com.adobe.acs.commons.fam.impl.ActionManagerImpl.withResolver(ActionManagerImpl.java:204)
        at com.adobe.acs.commons.fam.impl.ActionManagerImpl.runActionAndLogErrors(ActionManagerImpl.java:166)
        at com.adobe.acs.commons.fam.impl.ActionManagerImpl.lambda$deferredWithResolver$0(ActionManagerImpl.java:158)
        at com.adobe.acs.commons.fam.impl.TimedRunnable.run(TimedRunnable.java:84)
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
07.06.2018 16:51:11.244 *ERROR* [pool-647-thread-1] com.adobe.acs.commons.fam.impl.ActionManagerImpl Caught exception in task: in is null
java.lang.NullPointerException: in is null
        at java.util.zip.ZipInputStream.<init>(Unknown Source)
        at java.util.zip.ZipInputStream.<init>(Unknown Source)
        at org.apache.poi.openxml4j.opc.ZipPackage.<init>(ZipPackage.java:88)
        at org.apache.poi.openxml4j.opc.OPCPackage.open(OPCPackage.java:272)
        at org.apache.poi.util.PackageHelper.open(PackageHelper.java:37)
        at org.apache.poi.xssf.usermodel.XSSFWorkbook.<init>(XSSFWorkbook.java:258)
        at com.adobe.acs.commons.mcp.impl.processes.TagCreator.lambda$parseTags$0(TagCreator.java:131)
        at com.adobe.acs.commons.fam.impl.ActionManagerImpl.withResolver(ActionManagerImpl.java:204)
        at com.adobe.acs.commons.mcp.impl.processes.TagCreator.parseTags(TagCreator.java:130)
        at com.adobe.acs.commons.mcp.impl.ProcessInstanceImpl.lambda$runStep$14(ProcessInstanceImpl.java:238)
        at com.adobe.acs.commons.fam.impl.ActionManagerImpl.withResolver(ActionManagerImpl.java:204)
        at com.adobe.acs.commons.fam.impl.ActionManagerImpl.runActionAndLogErrors(ActionManagerImpl.java:166)
        at com.adobe.acs.commons.fam.impl.ActionManagerImpl.lambda$deferredWithResolver$0(ActionManagerImpl.java:158)
        at com.adobe.acs.commons.fam.impl.TimedRunnable.run(TimedRunnable.java:84)
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
`

**Comment by davidjgonzalez** [comment]
Can you attach or email me your input file (if you don’t want to share w the world)? Dgonzale@adobe.com

**Comment by Lrochefort** [comment]
Hi @davidjgonzalez 

I am using the sample file from ACS.

[acs-commons__tag-creator--sample-file.xlsx](https://github.com/Adobe-Consulting-Services/acs-aem-commons/files/2084685/acs-commons__tag-creator--sample-file.xlsx)

**Comment by Lrochefort** [comment]
@davidjgonzalez 

As you can see in the following screenshots, after selecting the Excel file, I kind of expect the see the file name selected on the UI. Maybe the file upload is broken and that's why we get a null pointer (cannot find the file).

![capture1](https://user-images.githubusercontent.com/5980343/41163362-74dbeb5e-6b06-11e8-8435-a7e4ddd5445b.PNG)
![capture2](https://user-images.githubusercontent.com/5980343/41163363-75020802-6b06-11e8-9bc1-1ea112af944b.PNG)
![capture3](https://user-images.githubusercontent.com/5980343/41163364-7527d1c2-6b06-11e8-8fb8-714410f98bd5.PNG)
![capture4](https://user-images.githubusercontent.com/5980343/41163365-755c54d8-6b06-11e8-8e45-e7237c467074.PNG)

**Comment by Lrochefort** [comment]
@davidjgonzalez You were saying that the file from @ksurendra is working on your instance.

Can you tell us what your setting looks like? AEM version, OS, etc.

Thanks

**Comment by ksurendra** [comment]
@Lrochefort  I tried to run the import again using the sample file 
[nfl1 (1).xlsx](https://github.com/Adobe-Consulting-Services/acs-aem-commons/files/2085744/nfl1.1.xlsx) and worked for me.

![screen shot 2018-06-08 at 2 20 53 pm](https://user-images.githubusercontent.com/902972/41176765-46ad58dc-6b27-11e8-90c2-087a7ecaea2f.png)

I'm on AEM 6.3 CFP 2.0 (aem-6.3.1-cfp-2.0.zip) and ACS Commons 3.17.0

**Comment by davidjgonzalez** [comment]
@Lrochefort - just ran it again on my machine and it was fine..

- macOS 10.13.4
- Google Chrome
- AEM 6.4.0
- acs-aem-commons-content-3.17.0-min.zip
- java version "1.8.0_121"
- Never opened Excel, so its just the files attached on this ticket.

[tag_creator_no_description.xlsx](https://github.com/Adobe-Consulting-Services/acs-aem-commons/files/2091700/tag_creator_no_description.xlsx)
[acs-commons__tag-creator--sample-file.xlsx](https://github.com/Adobe-Consulting-Services/acs-aem-commons/files/2091701/acs-commons__tag-creator--sample-file.xlsx)

**Comment by Lrochefort** [comment]
@davidjgonzalez 

I tried again with your files (without opening them) and it fails. I even tried FireFox and Edge. With those, the UI is not even displaying correctly...

I'm testing on a local instance with AEM **6.2 SP1**

I tried to downgrade to ACS Commons 3.11.0 to see if works with that version. It still fails.

So now, I reinstalled acs-aem-commons-content-3.17.0-min.zip, restarted the instance and retried. Still not working...

I realized that there's a bunch of javascript errors in the Chrome console when I load:

http://localhost:4502/etc/acs-commons/manage-controlled-processes.html

See screenshot.

![capture5](https://user-images.githubusercontent.com/5980343/41298223-e367a7c2-6e2e-11e8-8e61-d55301533496.PNG)

Do you know if 6.2 is supported? Do you know someone on 6.2 that could make it work?

I really have high hopes for this tool.

**Comment by davidjgonzalez** [comment]
[acs-aem-commons-content-3.17.1-SNAPSHOT-min.zip](https://github.com/Adobe-Consulting-Services/acs-aem-commons/files/2095450/acs-aem-commons-content-3.17.1-SNAPSHOT-min.zip)

@Lrochefort  can you try the attached build out? Make sure to clear your browser cache.

I was able to reproduce on AEM 6.2 (issue did not exhibit on 6.4, I assume not 6.3 but haven't tested).

It appears there is a bug in the larger MCP JS (that controls the process lists, forms, etc.) in that CoralUI render logic is being applied twice to all MCP forms. This is specifically bad for the FileUpload GraniteUI widget since it creates 2 hidden input fields, which results in the backend process to think that no file is being sent (since the file is referenced in the first input field, but the extra blank input field is superseding it).

I also cleaned up the JS issues you noted, which didn't have anything to do w the error, but still ugly.

Let me know if this works for you, and ill see about testing it on 6.3 and 6.4 across processes to make sure this change didnt break anything else (help w this would be appreciated if you can!)

**Comment by Lrochefort** [comment]
@davidjgonzalez 

I installed version 3.17.1 and it works!

Thank you so much!

When do you think this version will be available as a release?

**Comment by davidjgonzalez** [comment]
@Lrochefort, unfortunately, this is going to take some regression testing (6.2, 6.3, 6.4 x the various MCP tools) - so I doubt before the weekend. If you could help out it'd certainly expedite.

**Comment by Lrochefort** [comment]
@davidjgonzalez @badvision 

Hi guys!

I see you have merged the fix into master. 

When do you expect to release 3.17.1 ?

Thanks

**Comment by Lrochefort** [comment]
Hi!

Any plans on releasing 3.17.1 soon?

Thanks,

Luc

**Comment by davidjgonzalez** [comment]
This issued was resolved / release in 3.17.2
